dotCMS/src/main/java/com/dotcms/content/elasticsearch/util/DotReactiveElasticsearchClient.java

```diff
@@ -0,0 +1,874 @@
+                .map(response -> response.statusCode().is2xxSuccessful()) //
+                .next();
+    }
+
+    /*
+     * (non-Javadoc)
+     * @see org.springframework.data.elasticsearch.client.reactive.ReactiveElasticsearchClient#ping(org.springframework.http.HttpHeaders, org.elasticsearch.action.index.IndexRequest)
+     */
+    @Override
+    public Mono<IndexResponse> index(HttpHeaders headers, IndexRequest indexRequest) {
+        return sendRequest(indexRequest, DotReactiveElasticsearchClient.RequestCreator.index(), IndexResponse.class, headers).publishNext();
+    }
+
+    /*
+     * (non-Javadoc)
+     * @see org.springframework.data.elasticsearch.client.reactive.ReactiveElasticsearchClient#indices()
+     */
+    @Override
+    public Indices indices() {
+        return this;
+    }
+
+    /*
+     * (non-Javadoc)
+     * @see org.springframework.data.elasticsearch.client.reactive.ReactiveElasticsearchClient#ping(org.springframework.http.HttpHeaders, org.elasticsearch.action.update.UpdateRequest)
+     */
+    @Override
+    public Mono<UpdateResponse> update(HttpHeaders headers, UpdateRequest updateRequest) {
+        return sendRequest(updateRequest, DotReactiveElasticsearchClient.RequestCreator.update(), UpdateResponse.class, headers).publishNext();
+    }
+
+    /*
+     * (non-Javadoc)
+     * @see org.springframework.data.elasticsearch.client.reactive.ReactiveElasticsearchClient#ping(org.springframework.http.HttpHeaders, org.elasticsearch.action.delete.DeleteRequest)
+     */
+    @Override
+    public Mono<DeleteResponse> delete(HttpHeaders headers, DeleteRequest deleteRequest) {
+
+        return sendRequest(deleteRequest, DotReactiveElasticsearchClient.RequestCreator.delete(), DeleteResponse.class, headers) //
+                .publishNext();
+    }
+
+    /*
+     * (non-Javadoc)
+     * @see org.springframework.data.elasticsearch.client.reactive.ReactiveElasticsearchClient#count(org.springframework.http.HttpHeaders, org.elasticsearch.action.search.SearchRequest)
+     */
+    @Override
+    public Mono<Long> count(HttpHeaders headers, CountRequest countRequest) {
+        return sendRequest(countRequest, DotReactiveElasticsearchClient.RequestCreator.count(), CountResponse.class, headers) //
+                .map(CountResponse::getCount) //
+                .next();
+    }
+
+    /*
+     * (non-Javadoc)
+     * @see org.springframework.data.elasticsearch.client.reactive.ReactiveElasticsearchClient#ping(org.springframework.http.HttpHeaders, org.elasticsearch.action.search.SearchRequest)
+     */
+    @Override
+    public Flux<SearchHit> search(HttpHeaders headers, SearchRequest searchRequest) {
+
+        return sendRequest(searchRequest, DotReactiveElasticsearchClient.RequestCreator.search(), SearchResponse.class, headers) //
+                .map(SearchResponse::getHits) //
+                .flatMap(Flux::fromIterable);
+    }
+
+    public SearchResponse search(SearchRequest searchRequest, RequestOptions options){
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Parameter 'searchRequest' is not assigned and could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5113278)